### PR TITLE
Add missing dagger dependency to atlasdb-cli

### LINKED ISSUE #1742: Project cannot be compiled in IntelliJ
Trying to make in Intellij gives the following error:

```
Error:java: cannot access dagger.Module
  class file for dagger.Module not found
```

### PR DESCRIPTION
**Pull request by gsheasby**
Intellij complains of a missing dagger.Module, since the dependency was
shaded in atlasdb-dagger.

Caused by #1199.
Fixes #1742.
[no release notes]

**Goals (and why)**: Ensure devs can cleanly compile the project in intellij

**Implementation Description (bullets)**:
- Remove `"configuration: shadow"` from the dependency on `atlasdb-dagger` in `atlasdb-cli`
- Add `exclude module: dagger` from projects that depend on `atlasdb-cli`. 

**Concerns (what feedback would you like?)**:
- Do other projects pull in `atlasdb-cli`? They shouldn't, right?
- Will this cause any issues for the CLI distribution? Is there a sandbox where this can be easily tested?

**Where should we start reviewing?**:
- Check gradle and lock files
- Check out the branch and perform a clean compile of intellij

**Priority (whenever / two weeks / yesterday)**: Whenever

### CODE REVIEW

File: atlasdb-cli/build.gradle [hunk 1]
```diff
@@ -5,7 +5,7 @@ apply plugin: 'org.inferred.processors'
 
 dependencies {
     compile project(':atlasdb-cassandra')
-    compile project(path: ':atlasdb-dagger', configuration: 'shadow')
+    compile project(':atlasdb-dagger')
```
Comment (SerialVelocity) [inline atlasdb-cli/build.gradle]: Why don't we just add a `compileOnly` dependency on dagger with a message about why it is needed?

Comment (gsheasby) [inline atlasdb-cli/build.gradle]: That's much simpler, thanks for the tip!

File: atlasdb-cli/build.gradle [hunk 2]
```diff
@@ -6,6 +6,7 @@ apply plugin: 'org.inferred.processors'
 dependencies {
     compile project(':atlasdb-cassandra')
     compile project(path: ':atlasdb-dagger', configuration: 'shadow')
+    compileOnly('com.google.dagger:dagger')
```
Comment (SerialVelocity) [inline atlasdb-cli/build.gradle]: Another set of redundant parentheses here

**Comment by jboreiko** [comment]
@SerialVelocity mentioned that we might be able to reduce the compileOnly dep to just get the Module class but it appears that it's only in `dagger` so it doesn't appear possible.

**Pull request merged**